src/compute/src/server.rs

```diff
@@ -464,8 +442,15 @@ pub async fn compute_node_serve(
 
     // All set, let the meta service know we're ready.
     meta_client.activate(&advertise_addr).await.unwrap();
+    // Wait for the shutdown signal.
+    let _ = shutdown.cancelled().await;
+
+    // TODO(shutdown): gracefully unregister from the meta service.
 
-    (join_handle_vec, shutdown_send)
+    // NOTE(shutdown): We can't simply join the tonic server here because it only returns when all
+    // existing connections are closed, while we have long-running streaming calls that never
+    // close. From the other side, there's also no need to gracefully shutdown them if we have
+    // unregistered from the meta service.
```

Comment:
This is worth noting. However, this doesn't mean that passing shutdown token to the server is pointless, since it also rejects further incoming requests.